torchmetrics/metric.py

```diff
@@ -258,27 +269,47 @@ def sync(
                 only when running in a distributed setting.
             distributed_available: Function to determine if we are running inside a distributed setting
 
-        Returns:
-            cache: A dictionary containing the local metric states. The cache will be empty if sync didn't happen.
         """
+        if self.is_synced and should_sync:
+            raise MisconfigurationException("The Metric has already been synced.")
+
         is_distributed = distributed_available() if callable(distributed_available) else None
+
         if not should_sync or not is_distributed:
-            return {}
+            return
+
         if dist_sync_fn is None:
             dist_sync_fn = gather_all_tensors
+
         # cache prior to syncing
-        cache = {attr: getattr(self, attr) for attr in self._defaults}
+        self._cache = {attr: getattr(self, attr) for attr in self._defaults}
+
         # sync
         self._sync_dist(dist_sync_fn, process_group=process_group)
-        return cache
+        self.is_synced = True
+
+    def unsync(self, should_unsync: bool = True) -> None:
```

Comment:
missing docstring (public method)

Reply:
Done !

Reply:
similarly, why is `should_unsync` an argument here? can't the user make this determination before they call `unsync`?